api/v2/serializers/post/token_update.py

```diff
@@ -23,30 +24,61 @@ def validate(self, data):
         IF identity is found validation will:
         - Ensure that user/token produces a valid driver
         """
-        validated_data = data
-        self.validate_token_with_driver(data['provider'], data['username'], data['project_name'], data['token'])
-        return validated_data
+        logger.info("TokenUpdateSerializer:validate called")
+        # create will fail if it is an invalid token, but we need to create an identity first
```

Comment:
And I don't think we should create an identity before we are sure about the token whether is valid or not. If not, then why would we create an identity for that user? I understand that if keystone issues a token for that user, it must be valid for login purpose. However, given what we have so far. We either refactor the entire login process/logic, or we modify it to work with the current code base.